src/plugins/QuarterMaster.cpp

```diff
@@ -140,13 +137,11 @@ void QuarterMaster::OnStep(Builder* builder_) {
     gHistory.info() << "Request additional supplies: " <<
         expected_consumption << " >= " << expected_supply << std::endl;
 
-    m_skip_until_frame = gAPI->observer().GetGameLoop() + m_frames_to_skip;
-
+    m_next_run_turn = m_supply_wait_turns + gAPI->observer().GetGameLoop();
```

Comment:
Please, don't extend the PR with irrelevant details. We are trying to fix the gas saturation, not supplies build. It'll take much more time to finish PR, it'll extend each review turn.

Reply:
Okay, I thought you would want me to refactor redundant code at the same time as making this new 'plugin frequency' functionality. I'll remove and wait until the next improvement for supplies build after the Army updates.